src/webapp/reports/mal-data-approval/data-approval-list/hooks/useDataApprovalActions.tsx

```diff
@@ -0,0 +1,183 @@
+    const [reloadKey, reload] = useReload();
+    const { saveMonitoring: saveMonitoringValue } = useDataApprovalMonitoring();
+
+    const [globalMessage, setGlobalMessage] = useState<GlobalMessage>();
+    const [selectedIds, setSelectedIds] = useState<string[]>([""]);
+    const [revoke, { enable: enableRevoke, disable: disableRevoke }] = useBooleanState(false);
+    const [isDialogOpen, { enable: openDialog, disable: closeDialog }] = useBooleanState(false);
+
+    const activateMonitoringAction = useCallback(
+        async (selectedIds: string[]) => {
+            const items = _.compact(selectedIds.map(item => parseDataDuplicationItemId(item)));
+            if (items.length === 0) return;
+
+            saveMonitoringValue(items, true);
+            reload();
+        },
+        [reload, saveMonitoringValue]
+    );
+
+    const approveAction = useCallback(
+        async (selectedIds: string[]) => {
+            const items = _.compact(selectedIds.map(item => parseDataDuplicationItemId(item)));
+            if (items.length === 0) return;
+
+            const result = await compositionRoot.malDataApproval.updateStatus(items, "duplicate");
+            if (!result) setGlobalMessage({ type: "error", message: i18n.t("Error when trying to approve data set") });
+
+            saveMonitoringValue(items, true);
+            reload();
+        },
+        [compositionRoot.malDataApproval, reload, saveMonitoringValue]
+    );
+
+    const completeAction = useCallback(
+        async (selectedIds: string[]) => {
+            const items = _.compact(selectedIds.map(item => parseDataDuplicationItemId(item)));
+            if (items.length === 0) return;
+
+            const result = await compositionRoot.malDataApproval.updateStatus(items, "complete");
+            if (!result) setGlobalMessage({ type: "error", message: i18n.t("Error when trying to complete data set") });
+
+            reload();
+        },
+        [compositionRoot.malDataApproval, reload]
+    );
+
+    const deactivateMonitoringAction = useCallback(
+        async (selectedIds: string[]) => {
+            const items = _.compact(selectedIds.map(item => parseDataDuplicationItemId(item)));
+            if (items.length === 0) return;
+
+            saveMonitoringValue(items, false);
+            reload();
+        },
+        [reload, saveMonitoringValue]
+    );
+
+    const getDifferenceAction = useCallback(
+        async (selectedIds: string[]) => {
+            disableRevoke();
+            openDialog();
+            setSelectedIds(selectedIds);
+        },
+        [disableRevoke, openDialog, setSelectedIds]
+    );
+
+    const getDifferenceAndRevokeAction = useCallback(
+        async (selectedIds: string[]) => {
+            enableRevoke();
+            openDialog();
+            setSelectedIds(selectedIds);
+        },
+        [enableRevoke, openDialog, setSelectedIds]
+    );
+
+    const incompleteAction = useCallback(
```

Comment:
`completeAction` and `incompleteAction` are almost the same, can we have just one method `updateAction(selectedIds: [], status: "incomplete" | "complete")` and then pass the status to the usecase and generate the error message according the status value?